### PR TITLE
feat: make AppSync::GraphQLSchema property mappings

### PR DESCRIPTION
**Pull request by pradhapanr**
### Description of changes
Created properties for `Serverless::GraphQLApi` that map to the `AppSync::GraphQLSchema` resource.

Properties added:
- SchemaInline (maps to Definition)
- SchemaUri (maps to DefinitionS3Location)

### Description of how you validated changes
Created schema changes and ran against new transform tests.

### Checklist

- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [x] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [x] Using correct values
    - [x] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/sam_resources.py [hunk 1]
```diff
@@ -2137,23 +2137,31 @@ class SamGraphQLApi(SamResourceMacro):
...
+        appsync_schema = self._construct_appsync_schema(appsync_api.get_runtime_attr("api_id"))
+        resources: List[Resource] = [appsync_api, appsync_schema]
 
         return resources
 
     def _construct_appsync_api(self) -> GraphQLApi:
         api = GraphQLApi(logical_id=self.logical_id, depends_on=self.depends_on, attributes=self.resource_attributes)
 
+        if "Type" not in self.Auth:
+            raise InvalidResourceException(self.logical_id, "'Auth' object is missing required property 'Type'.")
```
Comment (aahung) [inline samtranslator/model/sam_resources.py:2163]: nit: Let's use `sam_expect(self.Auth, xxx, xxxx).to_be_a_str()

**Review by ssenchenko — DISMISSED** [review]

File: samtranslator/model/appsync.py [hunk 2]
```diff
@@ -23,3 +25,18 @@ class GraphQLApi(Resource):
...
     Tags: Optional[List[Dict[str, Any]]]
     XrayEnabled: Optional[bool]
+
+    runtime_attrs = {"api_id": lambda self: fnGetAtt(self.logical_id, "ApiId")}
+
+
+class GraphQLSchema(Resource):
+    resource_type = "AWS::AppSync::GraphQLSchema"
+    property_types = {
+        "ApiId": PropertyType(True, IS_DICT),
```
Comment (GavinZZ) [inline samtranslator/model/appsync.py:35]: Why is it `IS_DICT` on top but `str` below?

File: tests/translator/input/error_graphqlapi_no_schema_inline_or_uri.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Transform: AWS::Serverless-2016-10-31
+Resources:
+  SuperCoolAPI:
+    Type: AWS::Serverless::GraphQLApi
+    Properties:
+      SchemaInline: |
```
Comment (GavinZZ) [inline tests/translator/input/error_graphqlapi_no_schema_inline_or_uri.yaml:6]: nit: The file says `no_schema_inline_or_uri`, but the content is more like containing both`inline and uri`?

**Review by GavinZZ — COMMENTED** [review]
Left some comments, otherwise LGTM

File: tests/translator/input/error_graphqlapi_no_schema_inline_or_uri.yaml (hunk 3 above)
Comment (pradhapanr) [reply tests/translator/input/error_graphqlapi_no_schema_inline_or_uri.yaml:6]: Great catch! I think I accidentally changed the name when I was fixing the branch

File: samtranslator/model/appsync.py (hunk 2 above)
Comment (pradhapanr) [reply samtranslator/model/appsync.py:35]: The `Intrinsicable` type is a dictionary and then that inside type is what the specific values type is. `ApiId` is a string, but it's wrapped in a `GetAtt` intrinsic.

**Review by GavinZZ — APPROVED** [review]

**Review by ssenchenko — APPROVED** [review]

**Review by hoffa — APPROVED** [review]

**Pull request merged**